tests/functional/s/singleton_comparison.txt

```diff
@@ -1,8 +1,12 @@
 singleton-comparison:4::Comparison to None should be 'expr is None'
-singleton-comparison:5::Comparison to True should be just 'expr'
+singleton-comparison:5::Comparison to True should be just 'bool(expr)'
```

Comment:
I think `x = x==True` and bool(x) are not equivalent, for example if x = ["1"]:

```
>>> x = ["1"]
>>> y = x==True
False
>>> y = bool(x)
True
```

Reply:
Hmm, you're right about that. The current suggestion is incorrect as well though:
```
>>> x = ["1"]
>>> if x == True:
...     print("hello")
...
>>> if x: (suggested)
...     print("hello")
...
hello
>>> x = ["1"] == True
>>> x
False
>>> x = ["1"] (suggested)
>>> x
['1']
```
I think the suggestion should be `expr is True/False` instead of `expr/not expr`?

Reply:
Yeah, I think the problem if there is a comparison between a list and True is that the code is so wrong that we can't know what the fix is supposed to be. Maybe the spirit of the warning is that x==True is wrong whatever the value of x might be. I have a hard time thinking of a good message though...  "Don't mix boolean and other types" ? "Consider replacing comparison to 'True' by an appropriate call to bool()" ? "Suspicious comparison to True, consider using bool(x) or a typed boolean" ?

Reply:
How about `Use '{expr} is True' to check for the singleton value True, or '{expr}' to test for truthiness`?

Reply:
Very nice and imho understandable by everyone. Let's use that one :) !